src/plugins/mer/mervirtualboxmanager.cpp

```diff
@@ -540,6 +540,35 @@ bool MerVirtualBoxManager::addPortForwardingRule(const QString &vmName, const QS
     return true;
 }
 
+// It is an error to call this function when the VM vmName is running
+bool MerVirtualBoxManager::deletePortForwardingRule(const QString &vmName, const QString& ruleName) {
+    qCDebug(Log::vms) << "Deleting port forwarding rule" << ruleName << "from" << vmName;
+    QStringList arguments;
+    arguments.append(QLatin1String(MODIFYVM));
+    arguments.append(vmName);
+    arguments.append(QLatin1String(NATPF1));
+    arguments.append(QLatin1String(DELETE));
+    arguments.append(ruleName);
+    QTime timer;
+    timer.start();
+    VBoxManageProcess process;
+    if (!process.runSynchronously(arguments)) {
+        qWarning() << "VBoxManage failed to" << MODIFYVM;
+        return false;
+    }
+    qCDebug(Log::vms) << "Deleting port forwarding for rule" << ruleName
+                      <<  "took" << timer.elapsed() << "milliseconds";
+    return true;
+}
+
+// It is an error to call this function when the VM vmName is running
+bool MerVirtualBoxManager::updatePortForwardingRule(const QString& vmName, const QString& ruleName,
+                                                    quint16 newHostPort, quint16 newVmPort) {
+    if (deletePortForwardingRule(vmName, ruleName)) return false;
```

Comment:
Negate the condition? Up to one statement on a line place.

I would also prefer removing addPortForwardingRule, providing only updatePortForwardingRule, and implementing it like the other update* functions are implemented, achieving both deletion and addition with single vboxmanage invocation.

Reply:
Right, that's probably the reason why it is also split in case of QmlLive ports. So just omit the addPortForwardingRule and move the code under updatePortForwardingRule please.